google/cloud/spanner/value.cc

```diff
@@ -535,6 +560,22 @@ StatusOr<std::int64_t> Value::GetValue(std::int64_t,
   return x;
 }
 
+StatusOr<float> Value::GetValue(float, google::protobuf::Value const& pv,
+                                google::spanner::v1::Type const&) {
+  if (pv.kind_case() == google::protobuf::Value::kNumberValue) {
```

Comment:
Do we need to worry about loss of precision or anything like that?  If not, maybe a comment explaining why?

Reply:
I don't believe we have anything to worry about in practice.  We do have the (perhaps less than enlightening "narrowing" comment, due to the decision upstream to re-use `google.protobuf.Value { double number_value }` for the wire format.  We already assert that a C++ `double` can represent such a value, and now we also assert that `float -> double [-> proto-double -> double] -> float` is an unsurprising conversion path.  If it is surprising (or if it can't even be done in a language), they prescribe using the native 64-bit floating-point type to represent FLOAT32.

Reply:
Ack.  Just consider a more enlightening comment, I read it and thought "yes, it is narrowing, but why is it Okay in this case?"  Would be nice if we could succinctly answer that question.

Reply:
Done.